refactor: make RustScan additionally a library

PsypherPunk opened the pull request:
- add `lib.rs` (permitted as per [_Packages and Crates_](https://doc.rust-lang.org/book/ch07-01-packages-and-crates.html)).
- refactor `main.rs` to reference the library.
- update module references as required.

relates #497

By way of example, configuring a new binary project which references this crate:

```toml
[dependencies]
async-std = "^1.7"
rustscan = { path = "/home/rust/git/RustScan" }
```

…allows `rustscan` to be referenced as a library:

```rust
use async_std::task::block_on;
use std::{net::IpAddr, time::Duration};

use rustscan::input::{PortRange, ScanOrder};
use rustscan::port_strategy::PortStrategy;
use rustscan::scanner::Scanner;

fn main() {
    let addrs = vec!["127.0.0.1".parse::<IpAddr>().unwrap()];
    let range = PortRange {
        start: 1,
        end: 1_000,
    };
    let strategy = PortStrategy::pick(&Some(range), None, ScanOrder::Random);
    let scanner = Scanner::new(
        &addrs,
        10,
        Duration::from_millis(100),
        1,
        true,
        strategy,
        true,
        vec![9000],
    );

    let scan_result = block_on(scanner.run());

    println!("{:?}", scan_result);
}
```

PsypherPunk commented on src/input.rs:
```diff
@@ -161,6 +161,27 @@ impl Opts {
         opts
     }
 
+    pub fn default() -> Self {
```
This is the only major change (as other tests reference this, they no longer work).

PsypherPunk replied:
I could move this to a `impl Default for Opts` which resolves the `clippy` warning, unless there's a better way…?

PsypherPunk replied:
Added `impl Default for Opts`.

PsypherPunk commented:
Also, tests pass _except_ that as this is now a library, doc-tests are running and failing.

bee-san commented:
Wow this is cool!!!! A long awaited feature. I think some docs on how to use the library would be great too 🥳

PsypherPunk commented:
@bee-san, I've fixed the failing parts:

- fixed the doc-tests in `src/benchmark/mod.rs`.
- changed the indentation in `src/scripts/mod.rs` as this was being interpreted as code—this might need a review as I'm not sure if it looks as intended in the resulting doc. output.
- I've marked a couple in `src/scanner/mod.rs` as `compile_fail`; I don't think there's a particularly easy way to instantiate a `Scanner` for the examples—I suppose things like this are going to be raised if this is to be used as a library; should there be a sensible `Default` for `Scanner`?

Re. the "docs on how to use the library", I'll see what I can do…

bee-san commented:
Wow! Thank you so much!!
